yarn-project/simulator/src/avm/avm_memory_types.ts

```diff
@@ -232,15 +232,19 @@ export class TaggedMemory implements TaggedMemoryInterface {
   // Whether to track and validate memory accesses for each instruction.
   static readonly TRACK_MEMORY_ACCESSES = process.env.NODE_ENV === 'test';
 
-  // FIXME: memory should be 2^32, but TS doesn't allow for arrays that big.
-  static readonly MAX_MEMORY_SIZE = Number((1n << 32n) - 2n);
+  // FIXME: memory should be 2^32, but TS max array size is: 2^32 - 1
+  static readonly MAX_MEMORY_SIZE = Number((1n << 32n) - 1n);
   private _mem: MemoryValue[];
 
   constructor() {
     // We do not initialize memory size here because otherwise tests blow up when diffing.
     this._mem = [];
   }
 
+  public getMaxMemorySize(): number {
```

Comment:
This one should be static (or just use TaggedMemory.MAX_MEMORY_SIZE).

Reply:
Removed and use TaggedMemory.MAX_MEMORY_SIZE